pilot/pkg/bootstrap/server.go

```diff
@@ -179,6 +179,14 @@ func NewServer(args *PilotArgs) (*Server, error) {
 		mux:            http.NewServeMux(),
 	}
 
+	if args.Config.ControllerOptions.WatchedNamespaces != "" {
+		// Add the control-plane namespace to the list of watched namespaces.
```

Comment:
Why do we need this? It seems reasonable just curious as it doesn't seem strictly required. I guess we need it to read the secrets, etc that reside in the same ns?

Reply:
IIRC we have some CRs that behave differently when deployed to the 'root namespace' (they might have been deprecated by now though?), plus by default we deploy the gateways to the same namespace as istiod

Reply:
That's the primary reason, yeah. Listing things like secrets and pods (ingress / egress gateways for instance) in the control-plane namespace. Probably doable to add the control-plane namespace directly to the pod cache or when needed for secrets listers etc., but that seemed a little error prone.

Reply:
regarding the gateways this might actually be argument to append the namespaces at install time, though

Reply:
I would remove this code - if user wants explicit control over watched namespaces, they should include the control plane ns. If they don't - they shouldn't. 

istio-system is messy, in particular with multiple revisions running. In many cases it's safer to set 'istio-config' to a different namespace or not use it at all. 

I wouldn't even add istio-system explicitly in installer either - let user do it.